examples/webserver/blocking/src/main/java/io/helidon/webserver/examples/blocking/Main.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2020 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.webserver.examples.blocking;
+
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import javax.json.Json;
+import javax.json.JsonBuilderFactory;
+import javax.json.JsonObject;
+
+import io.helidon.common.LogConfig;
+import io.helidon.common.configurable.ThreadPoolSupplier;
+import io.helidon.config.Config;
+import io.helidon.media.jsonp.JsonpSupport;
+import io.helidon.webclient.WebClient;
+import io.helidon.webserver.Routing;
+import io.helidon.webserver.ServerRequest;
+import io.helidon.webserver.ServerResponse;
+import io.helidon.webserver.WebServer;
+import io.helidon.webserver.blocking.BlockingHandler;
+import io.helidon.webserver.blocking.BlockingRequest;
+
+/**
+ * Main class of the example, starts the server.
+ */
+public final class Main {
+    private static final JsonBuilderFactory JSON = Json.createBuilderFactory(Map.of());
+
+    private static WebClient webClient;
+
+    private Main() {
+    }
+
+    /**
+     * Application main entry point.
+     * @param args command line arguments.
+     */
+    public static void main(final String[] args) {
+        startServer();
+    }
+
+    /**
+     * Start the server.
+     * @return the created {@link io.helidon.webserver.WebServer} instance
+     */
+    static WebServer startServer() {
+        // load logging configuration
+        LogConfig.configureRuntime();
+
+        // By default this will pick up application.yaml from the classpath
+        Config config = Config.create();
+
+        int clientPort = config.get("backend-port").asInt().orElse(-1);
+
+        // explicitly configure an executor service
+        // the default uses virtual threads if available, but not enforced
+        ThreadPoolSupplier executor = ThreadPoolSupplier.create(config.get("executor"));
+        BlockingHandler.executorService(executor);
+
+        WebServer server = WebServer.builder(createRouting(config))
+                .port(8081)
+                .config(config.get("server"))
+                .addMediaSupport(JsonpSupport.create())
+                .build();
+
+        server.start()
+                .thenAccept(ws -> {
+                    int backendPort;
+                    if (clientPort == -1) {
+                        backendPort = ws.port();
+                    } else {
```

Comment:
I would expect something like 
```java
        server.start();
        System.out.println("WEB server is up! http://localhost:" + server.port());
        server.whenShutdown().thenRun(() -> {
            System.out.println("WEB server is DOWN. Good bye!");
        });
```

Reply:
Maybe `server.startAndBlock()`

Reply:
You can use `server.start().await()` - I think blocking API is mostly needed for routing, not so much for starting/stopping the webserver.